bpf/lib/nat.h

```diff
@@ -403,7 +417,14 @@ snat_v4_rev_nat_handle_mapping(struct __ctx_buff *ctx,
 		return DROP_SNAT_NO_MAP_FOUND;
 
 	*state = __snat_lookup(map, tuple);
-	ret = snat_v4_track_connection(ctx, tuple, *state, has_l4_header, ct_action,
+
+	memcpy(&tuple_revsnat, tuple, sizeof(tuple_revsnat));
+	if (*state) {
+		tuple_revsnat.daddr = (*state)->to_daddr;
+		tuple_revsnat.dport = (*state)->to_dport;
+	}
+
+	ret = snat_v4_track_connection(ctx, &tuple_revsnat, *state, has_l4_header, ct_action,
```

Comment:
This chunk on its own (without the change to set `ostate->common.needs_ct = 1` for EgressGW connections) already fixes CT for replies of SNATed host connections, correct?

Then I would suggest to split this part off into its own patch, and describe the CT bug there in detail (as it actually pre-dates the EgressGW aspect). Right now the impact for such host connections isn't mentioned at all from what I can tell?